api/metadata/models.py

```diff
@@ -0,0 +1,49 @@
+from core.models import AbstractBaseExportableModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from organisations.models import Organisation
+
+from .fields import GenericObjectID
+
+FIELD_VALUE_MAX_LENGTH = 2000
+
+
+class FieldType(models.TextChoices):
+    INTEGER = "int"
+    STRING = "str"
+    BOOLEAN = "bool"
+    FLOAT = "float"
+
+
+class MetadataField(AbstractBaseExportableModel):
+    name = models.CharField(max_length=255)
+    type = models.CharField(
+        max_length=255, choices=FieldType.choices, default=FieldType.STRING
+    )
+    description = models.TextField(blank=True, null=True)
+    organisation = models.ForeignKey(Organisation, on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ("name", "organisation")
+
+
+class MetadataModelField(AbstractBaseExportableModel):
+    field = models.ForeignKey(MetadataField, on_delete=models.CASCADE)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    is_required = models.BooleanField(default=False)
```

Comment:
Just chatted to @dabeeeenster and @kyle-ssg about this. Ben made a good point that users may not want metadata to be required across the entire organisation. So, for example, on features, you may not want to make JIRA ticket number required across all the projects in your organisation. We need to think about a way to make things 'required with respect to'. 

Can you have a think about this @gagantrivedi ?